src/engine/sparqlExpressions/SparqlExpression.cpp

```diff
@@ -108,6 +108,13 @@ Estimates SparqlExpression::getEstimatesForFilterExpression(
   return {inputSizeEstimate, inputSizeEstimate};
 }
 
+// _____________________________________________________________________________
+std::optional<std::vector<PrefilterExprVariablePair>>
+SparqlExpression::getPrefilterExpressionForMetadata(
+    [[maybe_unused]] bool isNegated) const {
+  return std::nullopt;
+};
+
```

Comment:
You have forgotten one case where you can apply a filter
(I also didn't tell you this).
In the `RegexExpression` there is the special case of a `PrefixRegex` which is applied using binary search and can basically  be something like ` ?x begins with "hannes" -> ?x >= hannes && ?x <hannesA>" (but of course the logic is a little more involved. You can have a look at it, but maybe this can also be a separate PR as I am currently also refactoring the RegexExpression to not get conflicts.

Reply:
And while i am at it : The `startswith` can probably also be implemented in that way.